### ISSUE TITLE
InfluxDB is Missing from Data Sources

### ISSUE THREAD

**Issue by dufferzafar** [getredash/redash#4980]
### Issue Summary

I have self-hosted Redash instance and I am trying to add InfluxDB data source, but it is missing from  the "New Data Sources" list.

I've installed all dependencies mentioned in `requirements_all_ds.txt` - specifcally, `influxdb==5.2.3` is installed.

### Steps to Reproduce

1. `manage.py ds new`
2. Enter a name of the Data Source
3. Influx DB is missing from the choices

Here is a list of choices that I see:
<details> 
1. athena
2. bigquery
3. google_spreadsheets
4. graphite
5. mongodb
6. couchbase
7. mysql
8. rds_mysql
9. pg
10. redshift
11. cockroach
12. url
13. kibana
14. elasticsearch
15. aws_es
16. presto
17. hive
18. hive_http
19. databricks
20. impala
21. vertica
22. clickhouse
23. yandex_metrika
24. yandex_appmetrika
25. rockset
26. treasuredata
27. sqlite
28. dynamodb_sql
29. mssql
30. memsql
31. mapd
32. jirajql
33. google_analytics
34. axibasetsd
35. results
36. prometheus
37. qubole
38. db2
39. druid
40. kylin
41. drill
42. uptycs
43. snowflake
44. phoenix
45. json
46. Cassandra
47. scylla
48. dgraph
49. exasol
50. cloudwatch
51. cloudwatch_insights
</details>

### Technical details:

* Redash Version: v9 Beta
* Browser/OS: Firefox / RHEL 7
* How did you install Redash: Self Hosted

### Related

* This user has also experienced the exact same problem: https://discuss.redash.io/t/missing-data-source-type-influxdb/5384

**Comment by prince-mishra** [comment]
The `influxdb` query_runner imports `InfluxDBClusterClient` which is the trouble-maker here. Use `InfluxDBClient` instead and it'll all work.

The following patch should work

```diff --git redash/query_runner/influx_db.py redash/query_runner/influx_db.py                                                                     [0/1365]
index c9cfe756..1c1038c5 100644
--- redash/query_runner/influx_db.py
+++ redash/query_runner/influx_db.py
@@ -6,11 +6,12 @@ from redash.utils import json_dumps
 logger = logging.getLogger(__name__)

 try:
-    from influxdb import InfluxDBClusterClient
+    from influxdb import InfluxDBClient

     enabled = True

 except ImportError:
+    logger.error("Failed to import InfluxDB Client")
     enabled = False


@@ -68,7 +69,7 @@ class InfluxDB(BaseQueryRunner):
         return "influxdb"

     def run_query(self, query, user):
-        client = InfluxDBClusterClient.from_DSN(self.configuration["url"])
+        client = InfluxDBClient.from_dsn(self.configuration["url"])

         logger.debug("influxdb url: %s", self.configuration["url"])
         logger.debug("influxdb got query: %s", query)
@@ -82,7 +83,9 @@ class InfluxDB(BaseQueryRunner):
             error = None
         except Exception as ex:
             json_data = None
-            error = ex.message
+            import traceback
+            traceback.print_exc()
+            error = str(ex)

         return json_data, error

**Comment by gabrieldutra** [comment]
Thanks for pointing it out @prince-mishra, this has been fixed in #4531 and it's already part of v9-beta.

You need to make sure `influxdb` is updated to 5.2.3 for both server and worker. Assuming you have the `docker-compose` setup, make sure you followed [the upgrade instructions](https://github.com/getredash/redash/blob/master/CHANGELOG.md), specifically in this case, "force a recreation of your containers with `docker-compose up --force-recreate --build` and you should be good to go.".

I'll close this issue as the InfluxDB is working in our [preview environment](http://redash-preview.netlify.app). Let me know if the above will work or not in your Forum thread 🙂.
